designs/disruption-controls.md

```diff
@@ -0,0 +1,118 @@
+# Disruption Controls - (in v1beta1+)
+
+## Motivation
+Users have been asking for more control around the speed of Consolidation and the ability to restrict disruptions to their nodes. The demand for this is reflected by the +1s in [#1738](https://github.com/aws/karpenter/issues/1738). This is increasingly important as clusters can run at a highly varied scales, where some users have reported scale-down behaving too slow at higher scales, or too quick at smaller scales. Additionally, users that want to control when nodes can be disrupted must add PDBs to block node disruptions, add `do-not-evict` annotations on pods for sensitive workloads, or add `do-not-consolidate` annotations on nodes that shouldn't be consolidated. Natively integrating a better way to ensure nodes aren't disrupted will ease the burden that users have to take on with disruption.
+
+These controls will include a new `Disruption` block in the `v1beta1` `NodePool` that will contain all scale-down behavioral fields. Initially, this will include `Budgets`, which define a (1) parallelism of how many nodes can be deprovisioned at a time and a (2) cron schedule that determines when the parallelism applies, and a `ConsolidateAfter` field that will allow users to affect the speed at which Karpenter will scale down underutilized nodes. Future scale-down behavioral fields should be colocated with these fields.
```

Comment:
> a `ConsolidateAfter` field that will allow users to affect the speed at which Karpenter will scale down underutilized nodes

can you add more information on this? Is it an interval where consolidation is done every Nth minute?

Reply:
I've intentionally left some information out of this because the exact implementation of this is in-flux. The TLDR is that when Karpenter detects node(s) as underutilized, which is done one batch at a time, Karpenter will wait for the `consolidateAfter` before consolidating those node(s).

Reply:
nit: Should it just be "Budget"? Can you set multiple budgets within a nodepool?

Reply:
Yeah, you can use multiple budgets.